server/src/services/initialization/onInitialize.ts

```diff
@@ -30,6 +31,11 @@ export const onInitialize = (serverState: ServerState) => {
       extensionVersion,
     });
 
+    // fetch available solidity versions
+    await getAvailableSolcVersions(serverState);
```

Comment:
I would change the name to `setAvailableSolcVersions` or alter it to `serverState.solcVersions = await getAvailableSolcVersions();`